src/nodes/chunk_append/chunk_append.c

```diff
@@ -195,6 +195,80 @@ ts_chunk_append_path_create(PlannerInfo *root, RelOptInfo *rel, Hypertable *ht,
 			break;
 	}
 
+	if (!ordered || ht->space->num_dimensions == 1)
+	{
+		List *nested_children = NIL;
+		/*
+		 * Convert the sort nodes that refer to the same chunk into a single
+		 * mergeAppend node to combine compressed and uncompressed chunk output.
+		 *
+		 * NB: We assume that the sort nodes referring the same chunk appear
+		 * one after the other and so we iterate through the children examining
+		 * consecutive pairs. Is it possible that this assumption is wrong?
+		 */
+		List *group = NIL;
+		Oid relid = InvalidOid;
+
+		foreach (lc, children)
+		{
+			Path *child = (Path *) lfirst(lc);
+			/* Check if this is in new group */
+			if (child->parent->relid != relid)
```

Comment:
I understand you are referring to https://github.com/timescale/timescaledb/blob/main/tsl/src/nodes/decompress_chunk/decompress_chunk.c#L574 right? Initially I had attempted to add a MergeAppend node there as you suggest, in place of the Append node, but that didn't seem to produce the expected plans, and I couldn't figure out why, hence the current solution.

Reply:
Yes, it would be more straightforward and hence preferable to the current approach. I will revisit the solution with MergeAppend instead of Append in `decompress_chunk` to at least understand why it does not work, if it is not applicable.

Reply:
most likely in the current path candidates sorting is not considered properly
https://github.com/timescale/timescaledb/pull/5575 is a step in that direction - I guess extending that could eventually lead to some usefull path candidates

Reply:
it seems it is not possible to produce a MergeAppend node at the decompression level, because of accumulate_append_subpath that gets called afterwards :
```
* accumulate_append_subpath
*      Add a subpath to the list being built for an Append or MergeAppend.
*
* It's possible that the child is itself an Append or MergeAppend path, in
* which case we can "cut out the middleman" and just add its child paths to
* our own list.  (We don't try to do this earlier because we need to apply
* both levels of transformation to the quals.)
```
So I think that I should proceed by working at the chunk append level instead of the decompress chunk level